prisma/schema.prisma

```diff
@@ -76,3 +76,33 @@ model Authenticator {
 
   @@id([userId, credentialID])
 }
+
+model App {
+  id        String       @id @unique
+  createdAt DateTime     @default(now())
+  versions  AppVersion[]
+}
+
+model AppVersion {
+  id           Int                    @unique @default(autoincrement())
+  createdAt    DateTime               @default(now())
+  value        String
+  builtAt      DateTime?
+  appId        String
+  dependencies AppVersionDependency[] @relation("AppVersionDependency")
+  dependsOn    AppVersionDependency[] @relation("AppVersionDependsOn")
+
+  app App @relation(fields: [appId], references: [id])
+
+  @@id([appId, value])
+}
+
+model AppVersionDependency {
+  id                     Int      @id @unique @default(autoincrement())
+  createdAt              DateTime @default(now())
+  dependantAppVersionId  Int
+  dependencyAppVersionId Int
+
+  dependantAppVersion  AppVersion? @relation("AppVersionDependency", fields: [dependantAppVersionId], references: [id])
```

Comment:
We have to remove dependency when the version is removed. Or don't?

Why are version fields optional?